packages/hardhat-plugin/package.json

```diff
@@ -80,6 +81,7 @@
     "ink": "3.2.0",
     "ink-spinner": "4.0.3",
     "ndjson": "2.0.0",
+    "prompts": "^2.4.2",
```

Comment:
I think we should leverage the ink framework. I know it is more hassle, but better one ui approach than many.

Maybe leverage something like: https://github.com/kevva/ink-confirm-input

Reply:
Hmm I kind of disagree in this one instance, but first I think there's a decision to be made. At what point do we want to prompt the user?

My inclination was to prompt immediately because anything else would be a waste if the user cancelled. I could also see the case for not wanting compilation to run on a cancelled deployment, not to mention clearing the terminal for a cancellation. If we go this route, then I think ink would not be the best decision because of how we encapsulated the ui renderer, although I guess we could just import and call `render` directly, but I personally think that's less clean than keeping ink in one place.

Reply:
Nice move. I am forced to agree that we should short circuit, which would imply the choice is between:

1. an early prompt using the prompt library
2. rewriting the prompt in ink out of a sense of consistency
3. refactoring the existing ink rendering code to an architecture that can support this use case

Lets go with the prompt library (1), your approach. I suspect more edge cases will push us to 3, but lets wait for them to appear as there is no difference in cost to doing them now or doing them later when we have more info.